pkg/controller/machinepool/openstackactuator.go

```diff
@@ -139,37 +138,30 @@ func (a *OpenStackActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment, po
 }
 
 // Get the OS image from an existing master machine.
-func getOpenStackOSImage(masterMachine *machinev1beta1.Machine, scheme *runtime.Scheme, logger log.FieldLogger) (string, error) {
-	providerSpec, err := decodeOpenStackMachineProviderSpec(masterMachine.Spec.ProviderSpec.Value, scheme)
-	if err != nil {
+func getOpenStackOSImage(masterMachine *machinev1beta1.Machine, logger log.FieldLogger) (string, error) {
+	var u *unstructured.Unstructured
+	if err := json.Unmarshal(masterMachine.Spec.ProviderSpec.Value.Raw, u); err != nil {
```

Comment:
I believe the second argument passed to `Unmarshal` cannot be nil. 
```suggestion
	var u unstructured.Unstructured
	if err := json.Unmarshal(masterMachine.Spec.ProviderSpec.Value.Raw, &u); err != nil {
```

Reply:
@fxierh to the rescue again!

Also had to change the line below.

Rebased, revised, rebuilt, and re-pushed `quay.io/2uasimojo/hive:openstack-providerspec-unstructured`

Thanks!

Reply:
I PTO-ed yesterday and I'll test it today.